src/vs/editor/browser/services/treeSitter/treeSitterParserService.ts

```diff
@@ -80,13 +84,12 @@ class TextModelTreeSitter extends Disposable {
 			}
 		}
 
-		this._treeSitterTree.tree = await this._treeSitterParser.parse(this.model, this._treeSitterTree);
+		treeSitterTree.tree = await this._treeSitterParser.parse(this.model, treeSitterTree);
 	}
 
 	override dispose() {
 		super.dispose();
-		this._treeSitterTree?.dispose();
-		this._unregisterModelListeners();
+		this._languageSessionDisposables.dispose();
```

Comment:
This is not needed, as the `this._register` call takes care of that :)
The nice thing about `_register` is that you can see at the creation location in the source that it is eventually going to be disposed.

Reply:
🤦‍♀️